qa/ingest-with-mustache/src/test/java/org/elasticsearch/ingest/IngestDocumentMustacheIT.java

```diff
@@ -44,6 +44,30 @@ public void testAccessMetaDataViaTemplate() {
         assertThat(ingestDocument.getFieldValue("field1", String.class), equalTo("2 bar"));
     }
 
+    // TODO(talevy): expected?
+    public void testAccessSubFieldsViaTemplate() {
+        Map<String, Object> document = new HashMap<>();
+        Map<String, Object> fooDoc = new HashMap<>();
+        fooDoc.put("bar", "value");
+        document.put("foo", fooDoc);
+        IngestDocument ingestDocument = new IngestDocument("index", "type", "id", null, null, null, null, document);
+        ingestDocument.setFieldValue(templateService.compile("field1"), ValueSource.wrap("1 {{foo.bar}}", templateService));
+        assertThat(ingestDocument.getFieldValue("field1", String.class), equalTo("1 value"));
+    }
+
+    // TODO(talevy): expected?
+    public void testAccessFieldWithDotsViaTemplate() {
+        Map<String, Object> document = new HashMap<>();
+        Map<String, Object> fooBarDoc = new HashMap<>();
+        fooBarDoc.put("val", "value");
+        document.put("foo.bar", fooBarDoc);
+        IngestDocument ingestDocument = new IngestDocument("index", "type", "id", null, null, null, null, document);
+        ingestDocument.setFieldValue(templateService.compile("field1"), ValueSource.wrap("1 {{foo.bar}}", templateService));
+        assertThat(ingestDocument.getFieldValue("field1", String.class), equalTo("1 {val=value}"));
+        ingestDocument.setFieldValue(templateService.compile("field2"), ValueSource.wrap("1 {{foo.bar.val}}", templateService));
+        assertThat(ingestDocument.getFieldValue("field2", String.class), equalTo("1 "));
+    }
+
```

Comment:
++ on discussing it separately, there are other problems related to supporting dots in field names in ingest although they are not supported in es core. None of the IngestDocument methods support accessing a field with a dot in its name at the moment, so with this PR the assumption is that dedot is the only processors that knows about dots and replaces them, and should run first for the following processors to work properly. We have to see whether this is reasonable or whether we want to add some way of escaping dots in field names to refer to them.